# E-Commerce Logs Analysis

## Project Objective

The objective of this project is to analyze the website logs from an e-commerce store to obtain useful insight into customer behaviour.

## Data Source

The dataset was obtained here: https://www.kaggle.com/datasets/kzmontage/e-commerce-website-logs and the data has an Open Database license: https://opendatacommons.org/licenses/odbl/1-0/

In [1]:
-- Specifying database
USE EcommerceLogsDb

Commands completed successfully.

Total execution time: 00:00:00.026

In [2]:
-- preview of table
SELECT TOP 3*
FROM logs;

(3 rows affected)

Total execution time: 00:00:00.378

access_date duration_seconds Proto IP Src_IP_type Src_Pt Bytes Accessed_From country membership Languages Sales Returned Returned_Amount 2016-11-01 09:58:00.0000000 2533 TCP 1.10.195.126 EXT_SERVER 8082 20100 Chrome Albania Normal Albanian 261.96 No 0 2016-11-01 09:59:00.0000000 4034 TCP 1.1.217.211 OPENSTACK_NET 56978 20500 Mozilla Firefox Albania Normal Albanian 731.94 No 0 2016-11-01 09:59:00.0000000 1525 TCP 1.115.198.107 EXT_SERVER 8082 90100 Mozilla Firefox Albania Normal Albanian 14.62 No 0

In [3]:
-- Which country had the most sales (transactions) and sales ($ amount)?
-- showing country rankings for both metrics using window functions
SELECT country, 
    COUNT(*) AS sales_transaction, 
    DENSE_RANK() OVER(ORDER BY COUNT(*) DESC) AS sales_transaction_rank, 
    SUM(Sales) AS sales_$, 
    DENSE_RANK() OVER(ORDER BY SUM(Sales) DESC) AS sales_$_rank
FROM logs
GROUP BY country
ORDER BY sales_$ DESC;

(102 rows affected)

Total execution time: 00:00:02.309

country sales_transaction sales_transaction_rank sales_$ sales_$_rank Austria 2674 1 1296463.3414999906 1 Belgium 2604 2 1154321.3705999977 2 Argentina 2603 3 1140238.6927999987 3 Australia 2520 4 1139576.1649999968 4 United States 2231 12 1132496.3800000001 5 Brazil 2604 2 1129919.2000999975 6 Norway 2160 13 1129793.889999999 7 Netherlands 2292 8 1106623.810000002 8 Spain 2232 11 1102578.460000004 9 Puerto Rico 2232 11 1089656.3800000006 10 Canada 2436 6 1078310.4730000005 11 United Kingdom 2232 11 1069649.1000000013 12 Bulgaria 2520 4 1068255.9588000018 13 Republic of Korea 2232 11 1063598.8800000008 14 Singapore 2232 11 1062442.5199999993 15 Suriname 2016 18 1037655.410000003 16 Mexico 2232 11 1034845.1700000012 17 Turkmenistan 2088 16 1032696.5800000039 18 Belize 2352 7 1009231.1078000019 19 Sweden 2148 14 1006956.3000000026 20 Chile 2604 2 998301.8032000033 21 Russian Federation 1944 20 989007.5500000018 22 Thailand 2004 19 984841.0600000018 23 Portugal 1944 20 955279.5800000016 24 Saint Lucia 2016 18 953605.8500000038 25 Mauritius 2292 8 951543.6300000029 26 Trinidad and Tobago 1944 20 940220.2900000017 27 Antigua and Barbuda 2268 9 937031.6729000005 28 Romania 2004 19 933676.0400000019 29 Armenia 2086 17 918236.707500002 30 Barbados 2099 15 914011.7824000027 31 Uruguay 2016 18 898636.5600000003 32 New Zealand 2088 16 897640.3300000007 33 Paraguay 1944 20 887757.4500000015 34 Luxembourg 2232 11 869798.3930000014 35 Ukraine 2016 18 860557.6200000015 36 Bahamas 1932 21 858137.5198000026 37 Malta 2232 11 849198.6836000021 38 Panama 1800 25 827407.4100000006 39 Poland 1728 29 823874.8300000004 40 Colombia 2514 5 823458.3767000027 41 Albania 1848 24 816714.1692000002 42 Guatemala 2160 13 811037.0116000031 43 Bahrain 1764 27 800315.182500001 44 Belarus 1763 28 793122.7560000019 45 Saint Vincent and Grenadines 1800 25 782619.9400000017 46 Greece 2232 11 781112.8694000007 47 Slovenia 1511 33 779910.2300000001 48 Slovakia 1584 30 774529.1199999993 49 Switzerland 1512 32 711132.8300000015 50 Ecuador 2232 11 704434.9145000005 51 Uzbekistan 1584 30 700289.7400000006 52 South Africa 1440 34 699497.384000001 53 Japan 2233 10 692880.2690000008 54 Israel 2232 11 678394.283000001 55 Grenada 1860 23 677864.1923000009 56 Iceland 2292 8 673693.6802000001 57 Guyana 1799 26 673308.8582000005 58 Kuwait 1800 25 668267.5795000002 59 Kyrgyzstan 1872 22 666682.2039000005 60 Ireland 2160 13 649537.6800000011 61 France 2160 13 643784.2059999998 62 Kazakhstan 1872 22 641065.9040000006 63 Hungary 1860 23 634635.7702 64 Costa Rica 2160 13 632901.6622000004 65 Finland 2088 16 630622.2634000012 66 Italy 2232 11 625664.3300000008 67 Seychelles 1296 36 620275.4700000003 68 Azerbaijan 1344 35 610523.8879999998 69 Germany 1872 22 601330.1426000008 70 El Salvador 1728 29 593740.2684 71 Lithuania 1572 31 572731.2999999988 72 Serbia 1296 36 564452.1499999986 73 Qatar 1068 40 554514.8799999999 74 Latvia 1511 33 549710.9939999995 75 Aruba 1176 38 537197.9709999991 76 Philippines 1079 39 522248.86000000004 77 Czech Republic 1932 21 517375.4399999993 78 Cuba 1728 29 500943.9629999991 79 Georgia 1584 30 499963.55559999857 80 Denmark 1584 30 449218.213999999 81 Estonia 1512 32 424996.7387999989 82 Croatia 1572 31 416275.25939999963 83 Sri Lanka 792 42 395918.2999999992 84 Montenegro 720 43 357511.14999999985 85 Jamaica 1224 37 347578.4049999999 86 Kiribati 792 42 323550.16689999966 87 Cyprus 1067 41 281314.48579999973 88 Turkey 504 44 280994.1200000001 89 Nicaragua 432 45 223247.22000000006 90 Maldives 720 43 217764.10799999998 91 United Arab Emirates 432 45 186636.71000000022 92 Fiji 792 42 186410.44499999998 93 Oman 216 46 105546.17000000007 94 San Marino 216 46 98001.99000000008 95 Saint Kitts and Nevis 216 46 87428.36000000006 96 Bosnia and Herzegovina 168 47 60497.63000000005 97 Cabo Verde 84 48 36562.07199999999 98 Macau 72 49 28242.788 99 Mongolia 60 50 22035.249999999985 100 Dominica 72 49 18743.108999999993 101 Russia 8 51 1264.5840000000003 102

From the table above, the top 4 countries for sales\_transaction and sales\_$ are Austria, Belgium, Argentina and Australia. The number of sales transactions seems highly correlated with sales ($) with some exceptions like the United States which is ranked 12th in number of transactions and 5th in sales ($).

Let's look at the average sale amount per country and compare to the worldwide average. The difference between the sale amount for each country and the worldwide average will also be computed.

In [4]:
-- average sales ($) per country
SELECT  
    country, AVG(Sales) AS avg_sales_$,
    -- subquery in SELECT to determine worldwide avg sales ($)
    (SELECT AVG(Sales) FROM logs) AS world_avg_sales_$,
    -- difference bewteen country avg and worldwide avg with another SELECT subquery
    (AVG(Sales) - (SELECT AVG(Sales) FROM logs)) AS difference
FROM logs
GROUP BY country
ORDER BY difference DESC;

(102 rows affected)

Total execution time: 00:00:00.371

country avg_sales_$ world_avg_sales_$ difference Turkey 557.5280158730159 411.3464487919592 146.18156708104272 Norway 523.0527268518517 411.3464487919592 111.70627805987857 Qatar 519.2086891385768 411.3464487919592 107.86224034660364 Nicaragua 516.7759722222225 411.3464487919592 105.42952343024939 Slovenia 516.1550165453345 411.3464487919592 104.80856775336139 Suriname 514.7100248015873 411.3464487919592 103.36357600961418 Russian Federation 508.7487397119337 411.3464487919592 97.40229091996054 United States 507.6182787987448 411.3464487919592 96.27183000677167 Sri Lanka 499.8968434343435 411.3464487919592 88.55039464237035 Montenegro 496.54326388888916 411.3464487919592 85.196815096916 Turkmenistan 494.5864846743296 411.3464487919592 83.24003588235644 Spain 493.98676523297456 411.3464487919592 82.6403164410014 Thailand 491.43765469061833 411.3464487919592 80.09120589864517 Portugal 491.39896090534944 411.3464487919592 80.05251211337628 Slovakia 488.9704040404043 411.3464487919592 77.62395524843112 Oman 488.63967592592604 411.3464487919592 77.29322713395288 Puerto Rico 488.19730286738286 411.3464487919592 76.8508540754097 South Africa 485.7620722222223 411.3464487919592 74.41562343024913 Austria 484.8404418474198 411.3464487919592 73.49399305544665 Philippines 484.01191844300297 411.3464487919592 72.66546965102981 Trinidad and Tobago 483.6524125514402 411.3464487919592 72.30596375946703 Netherlands 482.82016143106426 411.3464487919592 71.4737126390911 United Kingdom 479.2334677419356 411.3464487919592 67.88701894996245 Seychelles 478.6076157407409 411.3464487919592 67.26116694876777 Poland 476.77941550925925 411.3464487919592 65.43296671728609 Republic of Korea 476.52279569892477 411.3464487919592 65.17634690695161 Singapore 476.0047132616484 411.3464487919592 64.65826446967526 Saint Lucia 473.01877480158737 411.3464487919592 61.67232600961421 Switzerland 470.32594576719595 411.3464487919592 58.97949697522279 Sweden 468.7878491620108 411.3464487919592 57.44140037003763 Romania 465.90620758483027 411.3464487919592 54.55975879285711 Mexico 463.64030913978496 411.3464487919592 52.2938603478118 Panama 459.67078333333296 411.3464487919592 48.3243345413598 Aruba 456.80099574829933 411.3464487919592 45.45454695632617 Paraguay 456.6653549382713 411.3464487919592 45.31890614629816 Azerbaijan 454.25884523809526 411.3464487919592 42.9123964461221 San Marino 453.71291666666656 411.3464487919592 42.3664678746934 Bahrain 453.69341411564574 411.3464487919592 42.34696532367258 Australia 452.21276388888873 411.3464487919592 40.86631509691557 Belarus 449.87110380033994 411.3464487919592 38.52465500836678 Uruguay 445.752261904762 411.3464487919592 34.40581311278885 Bahamas 444.1705589026915 411.3464487919592 32.82411011071832 Belgium 443.28777672811117 411.3464487919592 31.94132793613801 Canada 442.65618760262726 411.3464487919592 31.309738810654096 Uzbekistan 442.10210858585833 411.3464487919592 30.755659793885172 Albania 441.94489675324695 411.3464487919592 30.59844796127379 Armenia 440.1901761744969 411.3464487919592 28.843727382523753 Argentina 438.04790349596607 411.3464487919592 26.70145470399291 Serbia 435.5340663580248 411.3464487919592 24.187617566051642 Barbados 435.451063554073 411.3464487919592 24.104614762099857 Cabo Verde 435.26276190476193 411.3464487919592 23.91631311278877 Saint Vincent and Grenadines 434.78885555555576 411.3464487919592 23.442406763582596 Brazil 433.9167435099843 411.3464487919592 22.570294718011155 United Arab Emirates 432.02942129629633 411.3464487919592 20.682972504323175 New Zealand 429.9043726053638 411.3464487919592 18.55792381339063 Belize 429.0948587585029 411.3464487919592 17.74840996652972 Ukraine 426.86389880952373 411.3464487919592 15.517450017550573 Bulgaria 423.91109476190434 411.3464487919592 12.564645969931178 Mauritius 415.15865183246086 411.3464487919592 3.8122030404877023 Antigua and Barbuda 413.1532949294533 411.3464487919592 1.806846137480136 Kiribati 408.52293800505066 411.3464487919592 -2

Turkey has the largest positive difference in average $ per sale with $146/sale greater than the worldwide average. In contrast, Russia has the largest negative difference with $253/sale less than the world average.

We'll now look at returns of purchased items for each country.

In [5]:
-- CTE for value of sales returned
WITH returned_$ AS (
    SELECT country,
        -- Sum of total dollars returned
        SUM(Returned_Amount) AS returned_amount_$,
        -- Percentage of dollars returned over dollars sold
        -- Sales column doesn't include amount for returned items
        SUM(Returned_Amount) / (SUM(Sales) + SUM(Returned_Amount)) AS pct_$_returned
    From logs
    GROUP BY country),

-- CTE for number of sales returned
returned_orders AS (
    SELECT t1.country, 
        -- counting all rows (number of transactions)
        COUNT(*) AS sales_transaction,
        -- counting number of order returns 
        COUNT(t2.Returned) AS orders_returned,
        -- calculating percentage of orders that were returned
        1.0 * COUNT(t2.Returned) / COUNT(*) AS pct_orders_returned
    FROM logs as t1
    LEFT JOIN (SELECT duration_seconds, ip, country, Returned
    FROM logs 
    WHERE Returned = 'Yes') AS t2
    ON t1.duration_seconds = t2.duration_seconds
    AND t1.ip = t2.ip
    GROUP BY t1.country)

-- final query with percent of orders returned and percent of sales refunded
SELECT TOP 5 CTE1.country, pct_orders_returned, pct_$_returned 
FROM returned_$ AS CTE1
    INNER JOIN returned_orders AS CTE2
    ON CTE1.country = CTE2.country 
ORDER BY pct_orders_returned DESC, pct_$_returned DESC;



Warning: Null value is eliminated by an aggregate or other SET operation.

(5 rows affected)

Total execution time: 00:00:00.534

country pct_orders_returned pct_$_returned Nicaragua 0.212962962962 0.2117912818013144 Oman 0.212962962962 0.15195254761361246 Montenegro 0.190277777777 0.21973417846937784 Puerto Rico 0.184139784946 0.15605512400495514 Poland 0.182870370370 0.14169463345436822

Nicaragua and Oman have the highest percentage of orders returned at 21.3%. This represents a possible opportunity to increase revenue by making the return policy more strict to attempt to lower the number of returned orders. The affect that a stricter return policy would have on customer experience should also be taken into account though.

We will next look at the number of returns for each language. This will help inform us of any language/communication issues thay may be causing a higher rate of purchase returns.

In [6]:
SELECT TOP 5 t1.Languages, 
        -- calculating percentage of orders that were returned
        ROUND((1.0 * COUNT(t2.Returned) / COUNT(*)), 2) AS pct_orders_returned
FROM logs as t1
LEFT JOIN (SELECT duration_seconds, ip, country, Returned
FROM logs 
WHERE Returned = 'Yes') AS t2
ON t1.duration_seconds = t2.duration_seconds
AND t1.ip = t2.ip
GROUP BY t1.Languages
ORDER BY pct_orders_returned DESC;

Warning: Null value is eliminated by an aggregate or other SET operation.

(5 rows affected)

Total execution time: 00:00:00.489

Languages pct_orders_returned marathi 0.220000000000 romanian 0.180000000000 kinyarwanda 0.180000000000 Slovak 0.170000000000 polish 0.170000000000

Marathi, a language spoken in India, has the highest percentage for orders returned (4% above the second language). A statistical test should be performed to determine if this percentage is statistically significantly greater than the 2nd highest percentage. If it is, this indicates that higher return rates could be linked to translation issues with the language. This means customers could be misinformed about products, resulting in higher return rates. The impact of language on return rates should be further investigated for the website as a whole to idenitfy if that is a frequent issue.

The membership types associated with online purchases will now be investigated per country. It is important to note that each count of a membership type represents an online purchase but not a unique individual (no unique identifiers are present in dataset). It is likely that premium members make more frequent purchases than normal and non-members. This category therefore investigates purchases per membership type and not individuals per membership type.

In [28]:
-- first CTE for total website purchase per country
WITH totals AS (
    SELECT country, COUNT(*) AS total_visits
    FROM logs
    GROUP BY country),

-- second CTE for website visits per membership type per country
memberships AS (
    SELECT country, membership, COUNT(1) AS website_visits
    FROM logs
    GROUP BY country, membership)

-- joining CTEs and calculating pct membership type for total visits per country
SELECT memberships.country, membership, website_visits, 1.0 * website_visits/total_visits AS pct_visits
FROM memberships
    LEFT JOIN totals
    ON memberships.country = totals.country
ORDER BY country;

(301 rows affected)

Total execution time: 00:00:00.094

country membership website_visits pct_visits Albania Normal 646 0.349567099567 Albania Premium 1185 0.641233766233 Albania Not Logged In 17 0.009199134199 Antigua and Barbuda Premium 1527 0.673280423280 Antigua and Barbuda Not Logged In 20 0.008818342151 Antigua and Barbuda Normal 721 0.317901234567 Argentina Not Logged In 24 0.009220130618 Argentina Premium 1612 0.619285439877 Argentina Normal 967 0.371494429504 Armenia Normal 678 0.325023969319 Armenia Premium 1388 0.665388302972 Armenia Not Logged In 20 0.009587727708 Aruba Normal 409 0.347789115646 Aruba Premium 763 0.648809523809 Aruba Not Logged In 4 0.003401360544 Australia Premium 1684 0.668253968253 Australia Not Logged In 41 0.016269841269 Australia Normal 795 0.315476190476 Austria Normal 835 0.312266267763 Austria Not Logged In 12 0.004487658937 Austria Premium 1827 0.683246073298 Azerbaijan Normal 424 0.315476190476 Azerbaijan Not Logged In 6 0.004464285714 Azerbaijan Premium 914 0.680059523809 Bahamas Normal 568 0.293995859213 Bahamas Not Logged In 10 0.005175983436 Bahamas Premium 1354 0.700828157349 Bahrain Not Logged In 11 0.006235827664 Bahrain Premium 1258 0.713151927437 Bahrain Normal 495 0.280612244897 Barbados Premium 1402 0.667937112910 Barbados Normal 687 0.327298713673 Barbados Not Logged In 10 0.004764173415 Belarus Not Logged In 14 0.007941009642 Belarus Premium 1203 0.682359614293 Belarus Normal 546 0.309699376063 Belgium Not Logged In 30 0.011520737327 Belgium Premium 1758 0.675115207373 Belgium Normal 816 0.313364055299 Belize Not Logged In 6 0.002551020408 Belize Normal 814 0.346088435374 Belize Premium 1532 0.651360544217 Bosnia and Herzegovina Not Logged In 2 0.011904761904 Bosnia and Herzegovina Normal 44 0.261904761904 Bosnia and Herzegovina Premium 122 0.726190476190 Brazil Not Logged In 16 0.006144393241 Brazil Normal 931 0.357526881720 Brazil Premium 1657 0.636328725038 Bulgaria Not Logged In 34 0.013492063492 Bulgaria Premium 1572 0.623809523809 Bulgaria Normal 914 0.362698412698 Cabo Verde Premium 57 0.678571428571 Cabo Verde Normal 27 0.321428571428 Canada Normal 854 0.350574712643 Canada Premium 1554 0.637931034482 Canada Not Logged In 28 0.011494252873 Chile Normal 857 0.329109062980 Chile Not Logged In 249 0.095622119815 Chile Premium 1498 0.575268817204 Colombia Not Logged In 579 0.230310262529 Colombia Premium 1087 0.432378679395 Colombia Normal 848 0.337311058074 Costa Rica Not Logged In 522 0.241666666666 Costa Rica Normal 691 0.319907407407 Costa Rica Premium 947 0.438425925925 Croatia Normal 429 0.272900763358 Croatia Premium 765 0.486641221374 Croatia Not Logged In 378 0.240458015267 Cuba Premium 926 0.535879629629 Cuba Normal 452 0.261574074074 Cuba Not Logged In 350 0.202546296296 Cyprus Not Logged In 258 0.241799437675 Cyprus Premium 517 0.484536082474 Cyprus Normal 292 0.273664479850 Czech Republic Normal 509 0.263457556935 Czech Republic Not Logged In 513 0.265527950310 Czech Republic Premium 910 0.471014492753 Denmark Premium 824 0.520202020202 Denmark Normal 370 0.233585858585 Denmark Not Logged In 390 0.246212121212 Dominica Normal 25 0.347222222222 Dominica Not Logged In 21 0.291666666666 Dominica Premium 26 0.361111111111 Ecuador Not Logged In 390 0.174731182795 Ecuador Premium 1152 0.516129032258 Ecuador Normal 690 0.309139784946 El Salvador Normal 559 0.323495370370 El Salvador Not Logged In 299 0.173032407407 El Salvador Premium 870 0.503472222222 Estonia Premium 683 0.451719576719 Estonia Not Logged In 471 0.311507936507 Estonia Normal 358 0.236772486772 Fiji Normal 212 0.267676767676 Fiji Not Logged In 264 0.333333333333 Fiji Premium 316 0.398989898989 Finland Not Logged In 507 0.242816091954 Finland Normal 550 0.263409961685 Finland Premium 1031 0.493773946360 France Normal 669 0.309722222222 France Not Logged In 406 0.187962962962 France Premium 1085 0.502314814814 Georgia Premium 901 0.568813131313 Georgia Not Logged In 223 0.140782828282 Georgia Normal 460 0.290404040404 Germany Premium 982 0.524572649572 Ge

In [32]:
SELECT country, Accessed_From, COUNT(1) AS website_visits
FROM logs
GROUP BY country, Accessed_From
ORDER BY country;

(700 rows affected)

Total execution time: 00:00:00.183

country Accessed_From website_visits Albania Safari 180 Albania Android App 392 Albania Others 281 Albania Mozilla Firefox 292 Albania Microsoft Edge 171 Albania IOS App 233 Albania Chrome 299 Antigua and Barbuda IOS App 289 Antigua and Barbuda Safari 210 Antigua and Barbuda Chrome 401 Antigua and Barbuda Others 336 Antigua and Barbuda Mozilla Firefox 357 Antigua and Barbuda Microsoft Edge 189 Antigua and Barbuda Android App 486 Argentina Others 407 Argentina IOS App 336 Argentina Chrome 429 Argentina Safari 240 Argentina Microsoft Edge 216 Argentina Android App 575 Argentina Mozilla Firefox 400 Armenia IOS App 250 Armenia Others 314 Armenia Safari 210 Armenia Chrome 304 Armenia Android App 487 Armenia Microsoft Edge 189 Armenia Mozilla Firefox 332 Aruba Mozilla Firefox 164 Aruba Safari 110 Aruba Android App 234 Aruba Chrome 228 Aruba Others 186 Aruba IOS App 155 Aruba Microsoft Edge 99 Australia Chrome 415 Australia Android App 575 Australia Mozilla Firefox 400 Australia Microsoft Edge 216 Australia Others 374 Australia Safari 240 Australia IOS App 300 Austria IOS App 346 Austria Chrome 418 Austria Mozilla Firefox 408 Austria Android App 576 Austria Others 421 Austria Safari 262 Austria Microsoft Edge 243 Azerbaijan IOS App 167 Azerbaijan Safari 128 Azerbaijan Mozilla Firefox 200 Azerbaijan Microsoft Edge 108 Azerbaijan Android App 298 Azerbaijan Others 205 Azerbaijan Chrome 238 Bahamas Android App 432 Bahamas Safari 186 Bahamas Others 292 Bahamas Chrome 296 Bahamas IOS App 243 Bahamas Microsoft Edge 177 Bahamas Mozilla Firefox 306 Bahrain IOS App 227 Bahrain Chrome 299 Bahrain Microsoft Edge 147 Bahrain Android App 373 Bahrain Safari 164 Bahrain Others 284 Bahrain Mozilla Firefox 270 Barbados Android App 482 Barbados Safari 211 Barbados Microsoft Edge 180 Barbados Others 311 Barbados Chrome 343 Barbados IOS App 252 Barbados Mozilla Firefox 320 Belarus Android App 410 Belarus Others 279 Belarus IOS App 220 Belarus Safari 170 Belarus Chrome 256 Belarus Microsoft Edge 153 Belarus Mozilla Firefox 275 Belgium Android App 556 Belgium Microsoft Edge 225 Belgium Chrome 458 Belgium Others 394 Belgium Mozilla Firefox 396 Belgium IOS App 325 Belgium Safari 250 Belize Safari 222 Belize Mozilla Firefox 359 Belize Android App 523 Belize Chrome 371 Belize Microsoft Edge 207 Belize Others 376 Belize IOS App 294 Bosnia and Herzegovina Safari 18 Bosnia and Herzegovina IOS App 18 Bosnia and Herzegovina Microsoft Edge 9 Bosnia and Herzegovina Mozilla Firefox 40 Bosnia and Herzegovina Android App 52 Bosnia and Herzegovina Others 15 Bosnia and Herzegovina Chrome 16 Brazil Others 403 Brazil Android App 551 Brazil IOS App 331 Brazil Mozilla Firefox 385 Brazil Safari 249 Brazil Microsoft Edge 225 Brazil Chrome 460 Bulgaria Others 382 Bulgaria Mozilla Firefox 399 Bulgaria IOS App 306 Bulgaria Chrome 401 Bulgaria Android App 576 Bulgaria Safari 240 Bulgaria Microsoft Edge 216 Cabo Verde Microsoft Edge 12 Cabo Verde Chrome 14 Cabo Verde IOS App 28 Cabo Verde Mozilla Firefox 12 Cabo Verde Others 18 Canada Others 376 Canada Safari 250 Canada IOS App 296 Canada Chrome 377 Canada Android App 552 Canada Microsoft Edge 213 Canada Mozilla Firefox 372 Chile Android App 575 Chile Chrome 457 Chile Safari 240 Chile IOS App 320 Chile Microsoft Edge 216 Chile Mozilla Firefox 400 Chile Others 396 Colombia Chrome 401 Colombia Android App 537 Colombia IOS App 326 Colombia Safari 240 Colombia Mozilla Firefox 386 Colombia Microsoft Edge 225 Colombia Others 399 Costa Rica Microsoft Edge 180 Costa Rica Others 336 Costa Rica Android App 483 Costa Rica IOS App 270 Costa Rica Chrome 357 Costa Rica Safari 210 Costa Rica Mozilla Firefox 324 Croatia Chrome 255 Croatia Others 240 Croatia Mozilla Firefox 246 Croatia Safari 150 Croatia Microsoft Edge 138 Croatia IOS App 198 Croatia Android App 345 Cuba Mozilla Firefox 255 Cuba Others 270 Cuba IOS App 198 Cuba Chrome 258 Cuba Safari 180 Cuba Android App 411 Cuba Microsoft Edge 156 Cyprus Microsoft Edge 84 Cyprus IOS App 150 Cyprus 